dev/release/utils-prepare.sh

```diff
@@ -40,6 +40,15 @@ update_versions() {
     meson.build
   rm -f meson.build.bak
   git add meson.build
+
+  glib_version_tuple="(${major_version}, 0)"
+  if [ "${type}" = "snapshot" ] && ! grep -q "${glib_version_tuple}" tool/generate-version-header.py; then
```

Comment:
Can we remove this condition?

```suggestion
  if [ "${type}" = "snapshot" ]; then
```

Reply:
This is needed otherwise duplicate entries get added for minor and patch releases. The `test_version_post_tag[release_type: :minor]` and `test_version_post_tag[release_type: :patch]` tests also fail due to the added duplicate versions.

Reply:
Ah, I see. How about this?

```diff
diff --git a/dev/release/utils-prepare.sh b/dev/release/utils-prepare.sh
index e4b12dad0a..d6891ad422 100644
--- a/dev/release/utils-prepare.sh
+++ b/dev/release/utils-prepare.sh
@@ -41,8 +41,11 @@ update_versions() {
   rm -f meson.build.bak
   git add meson.build
 
-  glib_version_tuple="(${major_version}, 0)"
-  if [ "${type}" = "snapshot" ] && ! grep -q "${glib_version_tuple}" tool/generate-version-header.py; then
+  # Add a new major version entry only when the next release is a new
+  # major release
+  if [ "${type}" = "snapshot" -a \
+       "${next_version}" = "${major_version}.0.0" ] ; then
+    local glib_version_tuple="(${major_version}, 0)"
     sed -i.bak -E -e \
       "s/^ALL_VERSIONS = \[$/&\\n        ${glib_version_tuple},/" \
       tool/generate-version-header.py
```

Reply:
:+1: Yeah that's a nicer solution